kong/conf_loader.lua

```diff
@@ -1041,6 +1092,28 @@ local function load(path, custom_conf, opts)
     -- find dynamic keys that need to be loaded
     local dynamic_keys = {}
 
+    local function add_dynamic_keys(t)
+      t = t or {}
+
+      for property_name, v_schema in pairs(CONF_INFERENCES) do
+        local directives = v_schema.directives
+        if directives then
+          local v = t[property_name]
+          if v then
+            for _, directive in ipairs(directives) do
+              dynamic_keys[directive] = true
+
+              if type(v) == "boolean" then
+                v = v and "on" or "off"
+              end
```

Comment:
can this be just before the loop?  i know, there's no performance constraints, and the compiler would CSE this away; and given that it's inside the `if v then ...` block, then `v` can't be `false`... so maybe the condition should be `if v ~= nil then ...` ?